### ISSUE TITLE
dbx_mdbx cannot open some profiles after updating libmdbx to 0.4

### ISSUE THREAD

**Issue by dartraiden** [miranda-ng/miranda-ng#2140]
https://forum.miranda-ng.org/index.php?topic=5935

1) both affected users use Windows 10 64-bit
2) the problem occurs only when opening an existing profile, a new profile is created succesfully

**Comment by unitwobble** [comment]
Posting a follow up from the topic since it'll be easier to track here:
1. Using Windows 7 64-bit
2. Existing profile from 0.95.11 alpha build #21986 (c311534fbf). 4MB in size.
3. Import existing profile into a new profile (MDBX and SQLite) works for all new versions.

However using existing profile on updated program:
1. Update to #22354, on program start, "Miranda can't understand that profile" error dialog and exits.
1. Update to #22355, existing profile is now 2MB and blanked out, hex editor shows mostly 0x00. Existing data is destroyed. Database Editor++ shows blank. Console appears with infinite scrolling of:
[xx:xx:xx 0B2C] txn_ptr_ro::~txn_ptr_ro failed with error=-30782, retrying...
[xx:xx:xx 0B2C] txn_ptr_ro::txn_ptr_ro failed with error=-30782, retrying...

**Comment by dartraiden** [comment]
Try [this build](https://github.com/miranda-ng/miranda-ng/commit/dbbad2e65220a724a43ea49825bc4a212b638db7)

**Comment by georgehazan** [comment]
@unitwobble that's not build 22355 that nullifies data in your profile, that was build 22354 that existed at the server for about 3 hours, now that problem completely gone

**Comment by unitwobble** [comment]
I just tested it.
![now](https://user-images.githubusercontent.com/24463506/70710799-f67d8d80-1d1a-11ea-9512-39b50fecad11.png)

**Comment by georgehazan** [comment]
@unitwobble yes, you need to restore your previous profile to avoid this problem
otherwise, you can try to use newly provided mdbx_chk.exe with the following command line:
>mdbx_chk -v -n -w <full_path_to_your_dat_file>. Post its output here if it doesn't help

**Comment by unitwobble** [comment]
@georgehazan 
I ran the tool on the existing profile. It didn't output anything useful:
```
C:\Miranda>mdbx_chk.exe -v -n -w C:\Miranda\Profiles\START2017-12-06\START2017-1
2-06.dat
 - monopolistic mode

C:\Miranda>
```
When Miranda is run it still wipes the profile like above. Sadly I cannot upload the profile since it contains alot of personal data.

**Comment by georgehazan** [comment]
@unitwobble that means that you still have the old Libs/libmdbx.mir. the safest way to upgrade your binaries is to make a new temporary profile, launch Plugin Updater in it, then remove that profile, restore yours from the backup and launch upgraded Miranda. 

Otherwise you can manually unzip this archive: https://miranda-ng.org/distr/x32/miranda32.zip (for 32-bit Miranda), remember, that the archive should be unzipped completely, not only Libs/libmdbx.mir. Then restore a backup and proceed

**Comment by unitwobble** [comment]
@georgehazan I used the existing installation of 21986 to create a blank profile with one Jabber account and tested it on a newly extracted distribution. It broke the same way.

Attaching the blank profile and VersionInfo.txt here. I hope this will help:
[BlankProfile.zip](https://github.com/miranda-ng/miranda-ng/files/3955996/BlankProfile.zip)

I mentioned above that the existing profile is erased when run in 22355. That is incorrect. It is just unreadable in the new version (shows nothing in the accounts, database editor shows nothing, but the proper database profile file is locked meaning it is using it). On my most recent test the 4MB file bloated up to 30MB+ (???) and when I moved it back to the old 21986 install, I can see the accounts and chat data intact. Don't know if it's the plugins or mdbx incompatibility between versions. I can try different setups (without plugins, etc) and post the results here if you want.
